### PR TITLE
A0-2431: Add rolling update partition input to updatenets

### PR DESCRIPTION
**Pull request by mikogs**

### CODE REVIEW

File: .github/actions/create-featurenet/action.yml [hunk 1]
```diff
@@ -129,7 +141,10 @@ runs:
         pr_image="${{ inputs.ecr-public-registry }}feature-env-aleph-node:${pr_image_tag}"
         ./Ops.sh create-featurenet \
           "${{ env.APP_NAME }}" \
-          "${pr_image}"
+          "${pr_image}" \
+          "false" \
+          "false" \
+          "${{ inputs.update == 'true' && inputs.rolling-update-partition || '0' }}"
```
Comment (Marcin-Radecki) [inline .github/actions/create-featurenet/action.yml:147]: We have in total three added args, but Ops.sh will have in total added 2?

File: .github/actions/create-featurenet/action.yml [hunk 2]
```diff
@@ -47,6 +47,10 @@ inputs:
     description: "Set to 'true' if it is just featurenet update"
     required: false
     default: 'false'
+  rolling-update-partition:
+    description: "Partition value when doing rolling update"
```
Comment (Marcin-Radecki) [inline .github/actions/create-featurenet/action.yml:51]: Can we use this sentence from https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/#partitions
> all Pods with an ordinal that is greater than or equal to the partition will be updated when the StatefulSet's .spec.template is updated. All Pods with an ordinal that is less than the partition will not be updated, and, even if they are deleted, they will be recreated at the previous version

rephrased e.g. as 

> all `aleph-node-validator-N` with an ordinal `N` that is greater than or equal to the partition will be updated. Default value: 0.

Also, this information needs to be put in Notion.

File: .github/actions/create-featurenet/action.yml [hunk 3]
```diff
@@ -82,6 +86,14 @@ runs:
           echo "!!! Invalid feature net node image tag"
           exit 1
         fi
+        if [[
+          "${{ inputs.update }}" == "true" && \
+          ! "${{ inputs.rolling-update-partition }}" =~ ^[0-9]{1,2}$
```
Comment (Marcin-Radecki) [inline .github/actions/create-featurenet/action.yml:91]: Technically we have only ten validators on Devnet, so range 0-9 is sufficient

File: .github/workflows/_create-featurenet.yml [hunk 4]
```diff
@@ -37,7 +37,7 @@ jobs:
   create-featurenet:
     needs: [check-vars-and-secrets]
     name: Create featurenet based on the PR
-    runs-on: ubuntu-20.04
+    runs-on: self-hosted
```
Comment (Marcin-Radecki) [inline .github/workflows/_create-featurenet.yml:40]: Can we create a tiny machine that runs these jobs?

**Review by Marcin-Radecki — APPROVED** [review]

File: .github/actions/create-featurenet/action.yml (hunk 1 above)
Comment (mikogs) [reply .github/actions/create-featurenet/action.yml:147]: That is not going to cause any problems.  There is an another PR in another repository to update that Ops.sh script where partition is actually last argument.  In that future version of Ops.sh the `create-featurenet` command goes as follows:

    create-featurenet <feature-net-name> <aleph-node-image|'testnet'|'mainnet'> [create-hook='false'] [generate-data-in-s3='false'] [rolling-update-partition='0']"

File: .github/actions/create-featurenet/action.yml (hunk 2 above)
Comment (mikogs) [reply .github/actions/create-featurenet/action.yml]: This is now updated and the description is as you suggested

Comment (mikogs) [reply .github/actions/create-featurenet/action.yml]: I will ensure that Notion is updated once all this is merged

File: .github/actions/create-featurenet/action.yml (hunk 3 above)
Comment (mikogs) [reply .github/actions/create-featurenet/action.yml]: It's now changed to ^[0-9]$

**Review by Marcin-Radecki — APPROVED** [review]
Rework looks good

**Pull request merged**